# HG changeset patch
# User Rik <rik@octave.org>
# Date 1424215445 28800
#      Tue Feb 17 15:24:05 2015 -0800
# Node ID ca6fc882177d3f43fc080706ddd1d6129d9e30e2
# Parent  3fa35defe495fe8daf6216bac5dcfa1dcac357bc
Fix working directory of Figure Save dialog between CLI and GUI (bug #44175).

* uiputfile.m: Check for a directory before calling fileparts which
automatically assumes the last element is a file.

* __add_default_menu__.m: Use default filename "untitled.pdf" for saving
FLTK figures.

diff --git a/scripts/gui/uiputfile.m b/scripts/gui/uiputfile.m
--- a/scripts/gui/uiputfile.m
+++ b/scripts/gui/uiputfile.m
@@ -89,17 +89,22 @@ function [retfile, retpath, retindex] = 
       outargs{2} = varargin{2};
     elseif (! isempty (varargin{2}))
       print_usage ();
     endif
   endif
 
   if (nargin > 2)
     if (ischar (varargin{3}))
-      [fdir, fname, fext] = fileparts (varargin{3});
+      if (isdir (varargin{3}))
+        fdir = varargin{3};
+        fname = fext = "";
+      else
+        [fdir, fname, fext] = fileparts (varargin{3});
+      endif
       if (! isempty (fdir))
         outargs{6} = fdir;
       endif
       if (! isempty (fname) || ! isempty (fext))
         outargs{3} = strcat (fname, fext);
       endif
     elseif (! isempty (varargin{3}))
       print_usage ();
diff --git a/scripts/plot/util/private/__add_default_menu__.m b/scripts/plot/util/private/__add_default_menu__.m
--- a/scripts/plot/util/private/__add_default_menu__.m
+++ b/scripts/plot/util/private/__add_default_menu__.m
@@ -71,17 +71,17 @@ function save_cb (h, e)
     __save_as__ (gcbo);
   endif
 endfunction
 
 function __save_as__ (caller)
   [filename, filedir] = uiputfile ({"*.pdf;*.ps;*.gif;*.png;*.jpg",
                                     "Supported Graphic Formats"},
                                    "Save Figure",
-                                   pwd);
+                                   [pwd, filesep, "untitled.pdf"]);
   if (filename != 0)
     fname = [filedir filesep() filename];
     obj = findall (gcbf, "label", "&Save");
     if (! isempty (obj))
       set (obj(1), "userdata", fname);
     endif
     saveas (caller, fname);
   endif
